src/product_logging/framework.rs

```diff
@@ -26,6 +31,65 @@ const SHUTDOWN_FILE: &str = "shutdown";
 /// File name of the Vector config file
 pub const VECTOR_CONFIG_FILE: &str = "vector.toml";
 
+/// Calculate the size limit for the log volume.
```

Comment:
I can't comment on the "meat" of this or the other PR but I really really like the extensive comment. Thank you!